### PR TITLE
Fix/sgrestart atconftest

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- added retry logic to verify ram on server cluster - addresses MB-40670
- added prepared statment logic to address - MB-40964

### CODE REVIEW

File: keywords/couchbaseserver.py [hunk 1]
```diff
@@ -199,10 +198,28 @@ def delete_buckets(self):
...
+        while count < 5:
+            resp = self._session.post("{}/query/service".format(query_url), data=verify_pdstmt_query_data)
+            resp_obj = resp.json()
+            status = resp_obj["status"]
+            result_count = resp_obj["metrics"]["resultCount"]
+            if status == "success":
+                if result_count == 0:
+                    break
+                count += 1
+                print("counting one more time for prepare statement check", count)
```
Comment (mgcouchbase) [inline keywords/couchbaseserver.py:220]: Can we import the logger for this

Comment (sridevi-15) [reply keywords/couchbaseserver.py]: forgot to remove the print statement. Removed it now.

**Review by mgcouchbase — COMMENTED** [review]
rest looks good

File: keywords/couchbaseserver.py [hunk 2]
```diff
@@ -337,20 +354,27 @@ def _get_mem_total_lowest(self, server_info):
                 # only use it if it's lower than previous low
                 mem_total_lowest = mem_total
 
-        if mem_total_lowest is None:
-            raise ProvisioningError("All nodes reported 0MB of RAM available")
+        """if mem_total_lowest is None:
```
Comment (mgcouchbase) [inline keywords/couchbaseserver.py:356]: we need this change I think

**Review by euniceatcouchbase — APPROVED** [review]
looks good to me. is maximum retry 5 times enough?

File: keywords/couchbaseserver.py (hunk 2 above)
Comment (sridevi-15) [reply keywords/couchbaseserver.py:356]: what change? I commented out the change, I did not remove. Once server team fix the ram issue, we can uncomment the change and revert the retries.

**Review by mgcouchbase — APPROVED** [review]

**Pull request merged**